src/server.rs

```diff
@@ -280,22 +280,41 @@ impl Server {
         function: FunctionInfo,
         is_const: bool,
     ) {
+
+        let second_route =  if route.ends_with("/") {
+             &route[0..route.len() - 1]
+        } else {
+             &format!("{}/", route)
+        };
+
+        self._add_route(py, route_type, route, &function, is_const);
+        self._add_route(py, route_type, second_route, &function, is_const);
+    }
+
+    fn _add_route(
+        &self,
+        py: Python,
+        route_type: &HttpMethod,
+        route: &str,
+        function: &FunctionInfo,
```

Comment:
```suggestion
        function: FunctionInfo,
```

Try this and remove the clones?

Reply:
you'd get a compiler error:

```
error[E0382]: use of moved value: `function`
   --> src/server.rs:289:69
    |
280 |         function: FunctionInfo,
    |         -------- move occurs because `function` has type `FunctionInfo`, which does not implement the `Copy` trait
...
284 |         self.save_route(py, route_type, route, function, is_const);
    |                                                -------- value moved here
...
289 |             self.save_route(py, route_type, &format!("{}/", route), function, is_const);
    |                                                                     ^^^^^^^^ value used here after move
    |
note: consider changing this parameter type in method `save_route` to borrow instead if owning the value isn't necessary
   --> src/server.rs:298:19
    |
293 |     pub fn save_route(
    |            ---------- in this method
...
298 |         function: FunctionInfo,
    |                   ^^^^^^^^^^^^ this parameter takes ownership of the value
help: consider cloning the value if the performance cost is acceptable
    |
284 |         self.save_route(py, route_type, route, function.clone(), is_const);
    |                                                        ++++++++

For more information about this error, try `rustc --explain E0382`.
```